crates/noirc_evaluator/src/ssa_refactor/ir/instruction/call.rs

```diff
@@ -0,0 +1,340 @@
+                SimplifyResult::SimplifiedTo(output)
+            } else {
+                SimplifyResult::None
+            }
+        }
+        BlackBoxFunc::SHA256 => simplify_hash(dfg, arguments, acvm::blackbox_solver::sha256),
+        BlackBoxFunc::Blake2s => simplify_hash(dfg, arguments, acvm::blackbox_solver::blake2s),
+        BlackBoxFunc::Keccak256 => {
+            match (dfg.get_array_constant(arguments[0]), dfg.get_numeric_constant(arguments[1])) {
+                (Some((input, _)), Some(num_bytes)) if array_is_constant(dfg, &input) => {
+                    let input_bytes: Vec<u8> = to_u8_vec(dfg, input);
+
+                    let num_bytes = num_bytes.to_u128() as usize;
+                    let truncated_input_bytes = &input_bytes[0..num_bytes];
+                    let hash = acvm::blackbox_solver::keccak256(truncated_input_bytes)
+                        .expect("Rust solvable black box function should not fail");
+
+                    let hash_values =
+                        vecmap(hash, |byte| FieldElement::from_be_bytes_reduce(&[byte]));
+
+                    let result_array = make_constant_array(dfg, hash_values, Type::unsigned(8));
+                    SimplifyResult::SimplifiedTo(result_array)
+                }
+                _ => SimplifyResult::None,
+            }
+        }
+        BlackBoxFunc::HashToField128Security => match dfg.get_array_constant(arguments[0]) {
+            Some((input, _)) if array_is_constant(dfg, &input) => {
+                let input_bytes: Vec<u8> = to_u8_vec(dfg, input);
+
+                let field = acvm::blackbox_solver::hash_to_field_128_security(&input_bytes)
+                    .expect("Rust solvable black box function should not fail");
+
+                let field_constant = dfg.make_constant(field, Type::field());
+                SimplifyResult::SimplifiedTo(field_constant)
+            }
+            _ => SimplifyResult::None,
+        },
+
+        BlackBoxFunc::EcdsaSecp256k1 => {
+            simplify_signature(dfg, arguments, acvm::blackbox_solver::ecdsa_secp256k1_verify)
+        }
+        BlackBoxFunc::EcdsaSecp256r1 => {
+            simplify_signature(dfg, arguments, acvm::blackbox_solver::ecdsa_secp256r1_verify)
+        }
+
+        BlackBoxFunc::FixedBaseScalarMul | BlackBoxFunc::SchnorrVerify | BlackBoxFunc::Pedersen => {
+            // Currently unsolvable here as we rely on an implementation in the backend.
+            SimplifyResult::None
+        }
+        BlackBoxFunc::RANGE | BlackBoxFunc::RecursiveAggregation => SimplifyResult::None,
+    }
+}
+
+fn make_constant_array(dfg: &mut DataFlowGraph, results: Vec<FieldElement>, typ: Type) -> ValueId {
+    let result_constants = vecmap(results, |element| dfg.make_constant(element, typ.clone()));
+
+    let typ = Type::Array(Rc::new(vec![typ]), result_constants.len());
+    dfg.make_array(result_constants.into(), typ)
+}
```

Comment:
Should this live in `DataFlowGraph`?

Reply:
I think it's fine to keep this here. It seems more specific to this one portion of code that a general function that more places would need. We can always move it later if other passes want access to it.